test/gtest/ucp/test_ucp_atomic.cc

```diff
@@ -232,6 +265,13 @@ class test_ucp_atomic : public test_ucp_memheap {
                 }
             }
 
+            /* Skip test if ODP requested but not supported */
+            if ((get_variant_value() & ODP) &&
```

Comment:
This condition was introduced to avoid running device atomic + odp variant on bluefield device since IB devices don't support ODP there and device atomics disables SW atomic emulation so there is protocol to send message in that case.

Reply:
Let's discuss it: if ODP is not supported, maybe we should skipp all ODP tests? But on other hand, this test is on UCP level, so maybe the issue is missing fallback logic in UCP?